### PR TITLE
Add truncated interval mesh utility

### PR DESCRIPTION
**Pull request by valeriabarra**
This is a revamp of #821 so it will supersede it. 
Unfortunately the original branch was so outdated (279 commits behind `main`) that rebasing it was a big headache. But credits for the original formulation go to @ilopezgp . I only reworked the interface to meet the reviewer's suggestions.

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

File: src/Meshes/Meshes.jl [hunk 1]
```diff
@@ -4,7 +4,10 @@ import RootSolvers
 using DocStringExtensions
 
 export RectilinearMesh,
-    EquiangularCubedSphere, EquidistantCubedSphere, ConformalCubedSphere
+    EquiangularCubedSphere,
+    EquidistantCubedSphere,
+    ConformalCubedSphere,
+    truncate_mesh!
```
Comment (charleskawczynski) [inline src/Meshes/Meshes.jl:10]:
```suggestion
    truncate_mesh
```

File: src/Meshes/interval.jl [hunk 2]
```diff
@@ -280,3 +280,47 @@ function IntervalMesh(
...
+
+
+"""
+    truncate_mesh(
+        parent_mesh::AbstractMesh,
+        trunc_domain::IntervalDomain{CT},
+    )
+Constructs an `IntervalMesh`, truncating the given `parent_mesh` defined on a
+truncated `trunc_domain`. The truncation preserves the number of
+degrees of freedom covering the space from the `trunc_domain`'s `:bottom` to `z_top`,
```
Comment (charleskawczynski) [inline src/Meshes/interval.jl:292]:
```suggestion
degrees of freedom covering the space from the `trunc_domain`'s `z_bottom` to `z_top`,
```

File: src/Meshes/interval.jl [hunk 3]
```diff
@@ -280,3 +280,47 @@ function IntervalMesh(
...
+    new_nelems = length(trunc_faces) - 1
+
+    Δz_top = trunc_faces[end].z - trunc_faces[end - 1].z
+    Δz_bottom = trunc_faces[2].z - trunc_faces[1].z
+
+    new_stretch = GeneralizedExponentialStretching(Δz_bottom, Δz_top)
+    new_domain = IntervalDomain(
+        z_bottom,
+        Geometry.ZPoint{FT}(z_top),
+        boundary_tags = (:bottom, :top),
```
Comment (charleskawczynski) [inline src/Meshes/interval.jl:323]:
```suggestion
        boundary_names = trunc_domain.boundary_names,
```

**Review by charleskawczynski — APPROVED** [review]
I think this looks fine. The only remaining thing might be to bikeshed `truncate_mesh`. @simonbyrne, any reason to use `truncate` over `truncate_mesh`?

**Comment by valeriabarra** [comment]
@charleskawczynski I think we can merge as-is and then maybe if there is a suggestion for a better name we can rename it later? 
Unless @simonbyrne is strongly opposed to `truncate_mesh`. I personally think it's clearer

**Review by ilopezgp — APPROVED** [review]

**Comment by charleskawczynski** [comment]
Supersedes #821

**Pull request merged**